### PR TITLE
move Get/SetInterchainAccount to keeper.go, add tests

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

Another small PR, more housekeeping for interchain-accounts branch.

closes: #315

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/keeper/keeper_test.go [hunk 1]
```diff
@@ -76,3 +77,31 @@ func (suite *KeeperTestSuite) TestGetPort() {
...
+	retrievdAddress, err := suite.chainB.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainB.GetContext(), counterpartyPortID)
+	suite.Require().NoError(err, "GetInterchainAccountAddress failed")
+	suite.Require().Equal(expectedAddr.String(), retrievdAddress)
+
+	retrievdAddress, err = suite.chainA.GetSimApp().ICAKeeper.GetInterchainAccountAddress(suite.chainA.GetContext(), "invalid port")
+	suite.Require().Error(err)
+	suite.Require().Empty(retrievdAddress)
+}
+
+func (suite *KeeperTestSuite) TestSetInterchainAccountAddress() {
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/keeper/keeper_test.go:123]: Done

File: modules/apps/27-interchain-accounts/keeper/keeper.go [hunk 2]
```diff
@@ -149,3 +150,23 @@ func (k Keeper) IsActiveChannel(ctx sdk.Context, portId string) bool {
 func (k Keeper) AuthenticateCapability(ctx sdk.Context, cap *capabilitytypes.Capability, name string) bool {
 	return k.scopedKeeper.AuthenticateCapability(ctx, cap, name)
 }
+
+func (k Keeper) GetInterchainAccountAddress(ctx sdk.Context, portId string) (string, error) {
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/keeper/keeper.go]: Done

File: modules/apps/27-interchain-accounts/keeper/keeper_test.go [hunk 3]
```diff
@@ -76,3 +77,31 @@ func (suite *KeeperTestSuite) TestGetPort() {
 	port := suite.chainA.GetSimApp().ICAKeeper.GetPort(suite.chainA.GetContext())
 	suite.Require().Equal(types.PortID, port)
 }
+
+func (suite *KeeperTestSuite) TestGetInterchainAccountAddress() {
+	path := NewICAPath(suite.chainA, suite.chainB)
+	suite.coordinator.Setup(path)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/keeper/keeper_test.go]: Have updated this and looks much better I think! Definitely making more sense now, however, I had to update `ChanOpenConfirm` to be invoked on `path.EndpointB` which is correct I believe!

File: modules/apps/27-interchain-accounts/keeper/keeper_test.go [hunk 4]
```diff
@@ -76,3 +98,33 @@ func (suite *KeeperTestSuite) TestGetPort() {
...
 }
+
+func (suite *KeeperTestSuite) TestGetInterchainAccountAddress() {
+	suite.SetupTest()
+	path := NewICAPath(suite.chainA, suite.chainB)
+	suite.coordinator.SetupConnections(path)
+
+	if err := SetupICAPath(path, "testing"); err != nil {
+		suite.FailNow("failed to configure interchain accounts path")
+	}
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/keeper/keeper_test.go:109]: this looks good, but historically we have used

`suite.Require().NoError(err)` which I believe calls `FailNow`. Do you think you could update it? Or do you think `suite.FailNow` is more suitable?

File: modules/apps/27-interchain-accounts/keeper/keeper_test.go [hunk 5]
```diff
@@ -5,6 +5,7 @@ import (
 
 	"github.com/stretchr/testify/suite"
 
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/keeper/keeper_test.go:8]: super small nit: could you group this import in the middle, [see reasoning](https://github.com/cosmos/ibc-go/pull/337#discussion_r694058339)

**Review by colin-axner — APPROVED** [review]
Looks great! Nice work! Left 2 very small nits

File: modules/apps/27-interchain-accounts/keeper/keeper_test.go (hunk 5 above)
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/keeper/keeper_test.go]: No problemo!

File: modules/apps/27-interchain-accounts/keeper/keeper_test.go (hunk 4 above)
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/keeper/keeper_test.go]: Updated! I think either is probably fine, I'm just generally used to using: `if err := funcCall(); err != nil {}` if there is only one return val of type `error`

**Pull request merged**